### PR TITLE
fix: Adds podAntiAffinity in base manifests (#4549)

### PR DESCRIPTION
**Pull request by mnacharov**
For #4549

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by mnach** [comment]
Running in cluster with 4 nodes:  (After installation all pods can be on the same node - k8s scheduler responsibility. `kubectl pod delete` causes rescheduling )
Снимок экрана 2020-10-19 в 18 10 23

Scale up argocd-server:
Снимок экрана 2020-10-19 в 19 06 13

Scale up redis-server:
Снимок экрана 2020-10-19 в 19 08 06

File: manifests/base/dex/argocd-dex-server-deployment.yaml [hunk 1]
```diff
@@ -39,3 +39,18 @@ spec:
...
+              labelSelector:
+                matchLabels:
+                  app.kubernetes.io/name: argocd-dex-server
+              topologyKey: kubernetes.io/hostname
+          - weight: 5
+            podAffinityTerm:
+              labelSelector:
+                matchLabels:
+                  app.kubernetes.io/part-of: argocd
+              topologyKey: kubernetes.io/hostname
```
Comment (jessesuen) [inline manifests/base/dex/argocd-dex-server-deployment.yaml:56]: I'm pretty sure the way we run dex we cannot scale it. So I think this should be removed as to not give false impression.

**Comment by jessesuen** [comment]
@mnach this looks good, but I have a small request to remove dex anti-affinity since it can't be scaled (the way we run it)

File: manifests/base/dex/argocd-dex-server-deployment.yaml (hunk 1 above)
Comment (mnach) [reply manifests/base/dex/argocd-dex-server-deployment.yaml]: removed

**Comment by mnacharov** [comment]
> @mnach this looks good, but I have a small request to remove dex anti-affinity since it can't be scaled (the way we run it)

fixed. please review

**Review by jessesuen — APPROVED** [review]

**Pull request merged**